Sources/CovidCertificateSDK/CertLogic/CertLogic.swift

```diff
@@ -126,11 +128,27 @@ class CertLogic {
                                            isSwitzerlandOnly: isSwitzerlandOnly))
     }
 
-    private func createPayload(from holder: CertificateHolderType) -> CertLogicPayload? {
-        guard let certificate = holder.certificate as? DCCCert else {
-            return nil
+    func checkModeRules(holder: CertificateHolderType, modes: [CheckMode], validationClock: Date = Date()) -> Result<ModeResults, CertLogicValidationError> {
+        var results: [CheckMode: ModeCheckResult] = [:]
+        let external = externalJson(validationClock: validationClock)
+        for mode in modes {
+            guard let modeRule = modeRule,
+                  let payload = createPayload(from: holder, mode: mode.id),
+                  let json = try? JSONEncoder().encode(payload) else {
+                return .failure(.JSON_ERROR)
+            }
+
+            let context = JSON(["external": external, "payload": JSON(json)])
+            if let validationCode: String = try? applyRule(modeRule, to: context) {
+                results[mode] = .init(isValid: validationCode == "SUCCESS", code: validationCode)
+            } else {
+                return .failure(.TEST_COULD_NOT_BE_PERFORMED(test: "MODE_CHECK"))
```

Comment:
This means we will fail completely even if just one mode could not be evaluated, right? Shouldn't we just return a failure for that mode (still return the array?)

Reply:
Yes that would be better: https://github.com/admin-ch/CovidCertificate-SDK-iOS/pull/118